plugin/trino-phoenix5/src/main/java/io/trino/plugin/phoenix5/PhoenixClientModule.java

```diff
@@ -162,7 +167,24 @@ public static Properties getConnectionProperties(PhoenixConfig config)
         }
 
         PhoenixEmbeddedDriver.ConnectionInfo connectionInfo = PhoenixEmbeddedDriver.ConnectionInfo.create(config.getConnectionUrl());
-        connectionInfo.asProps().asMap().forEach(connectionProperties::setProperty);
+        ReadOnlyProps props = connectionInfo.asProps();
+        props.asMap().forEach(connectionProperties::setProperty);
+        if (connectionInfo.getKeytab() != null) {
+            Configuration conf = HBaseFactoryProvider.getConfigurationFactory().getConfiguration();
+            for (Map.Entry<String, String> entry : props) {
+                conf.set(entry.getKey(), entry.getValue());
+            }
+            conf.set("hadoop.security.authentication", "Kerberos");
+            conf.set(QueryServices.HBASE_CLIENT_KEYTAB, connectionInfo.getKeytab());
+            conf.set(QueryServices.HBASE_CLIENT_PRINCIPAL, connectionInfo.getPrincipal());
```

Comment:
Instead of fetching them from the connection URL, can we have dedicated config classes for the same ?

Reply:
IMHO it's more natural to a Phoenix user to include those in the JDBC URL (as they would do with any other client).